services:
  db:
    image: postgres:14.1
    environment:
      - POSTGRES_DB=bb_db
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  rabbitmq:
    image: "rabbitmq:alpine"

    healthcheck:
      test: rabbitmq-diagnostics -q ping
      interval: 30s
      timeout: 30s
      retries: 3

  django:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: python manage.py runserver 0.0.0.0:8000
    environment:
      - DATABASE_URL=postgres://postgres:postgres@db:5432/bb_db
      - CELERY_BROKER_URL=amqp://guest:guest@rabbitmq:5672//
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    depends_on:
      - db
      - rabbitmq
    restart: on-failure

  celery:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: celery -A bb.tasks worker -l info --without-gossip --without-mingle --without-heartbeat
    environment:
      - DATABASE_URL=postgres://postgres:postgres@db:5432/bb_db
      - CELERY_BROKER_URL=amqp://guest:guest@rabbitmq:5672//
    volumes:
      - .:/app
    depends_on:
      - db
      - rabbitmq
    restart: on-failure

  beats:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: celery -A bb.tasks beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler
    environment:
      - DATABASE_URL=postgres://postgres:postgres@db:5432/bb_db
      - CELERY_BROKER_URL=amqp://guest:guest@rabbitmq:5672//
    volumes:
      - .:/app
    depends_on:
      - db
      - rabbitmq
    restart: on-failure
